incubator/hnc/pkg/forest/forest.go

```diff
@@ -461,79 +461,70 @@ func (ns *Namespace) HasCritCondition() bool {
 	return ns.Parent().HasCritCondition()
 }
 
-// HasCondition returns true if there's a condition with the given object and code. If code is the
-// empty string, it returns true if there's _any_ condition for the given object.
-func (ns *Namespace) HasCondition(obj api.AffectedObject, code api.Code) bool {
-	if _, exists := ns.conditions[obj]; !exists {
-		// Nothing for this obj
-		return false
-	}
-	if code == "" {
-		// Something exists for this obj; we don't care what
-		return true
-	}
-	_, exists := ns.conditions[obj][code]
-	return exists
-}
-
 // ClearCondition clears conditions in the namespace for a single object. If `code` is non-empty, it
 // only clears conditions with that code, otherwise it clears all conditions for that object. It
 // should only be called by the code that also *sets* the conditions.
 //
 // It returns true if it made any changes, false otherwise.
 func (ns *Namespace) ClearCondition(obj api.AffectedObject, code api.Code) bool {
-	if !ns.HasCondition(obj, code) {
+	if _, hasObj := ns.conditions[obj]; !hasObj {
+		// No conditions for this object
 		return false
 	}
 
 	if code == "" {
 		delete(ns.conditions, obj)
-	} else {
-		delete(ns.conditions[obj], code)
```

Comment:
Was this a bug that you fixed by returning the right value?

Reply:
No, the logic's still exactly the same - line 492 got moved to line 484. But we no longer rely on the `HasCondition` method - it was only being used here, so I deleted it.

Reply:
Right, I was confirming that it was a previous bug here to not `return false` if the condition wasn't there already at the first place. Your code moving it out and `return false` when `!hasCode` fixed this bug seems.

Reply:
Didn't the old line 485 catch that case?

```
if !ns.HasCondition(obj, code) {
  return false
```

I needed to add a bunch of `return false` statements because that one line no longer caught them all.

Reply:
Update: I'm reverting this part of the change since I have another change which needs the `HasCondition` method. Please LMK what you think the bug is!

Reply:
Ah NVM, then there was no bug in the old one. I missed the return in line 485.